identity-wallet/src/state/search/reducers/delete_recent_search.rs

```diff
@@ -0,0 +1,26 @@
+use crate::error::AppError;
+use crate::state::search::actions::delete_recent_search::DeleteRecentSearch;
+use crate::state::search::SearchResults;
+use crate::state::{
+    actions::{listen, Action},
+    AppState,
+};
+
+pub async fn delete_recent_search(state: AppState, action: Action) -> Result<AppState, AppError> {
+    if let Some(deletion) = listen::<DeleteRecentSearch>(action) {
```

Comment:
Rename `deletion` to `recent_search` since it contains the same data as `AddRecentSearch` and it makes no sense to use a different name here